nifi-api/src/main/java/org/apache/nifi/components/resource/ResourceReference.java

```diff
@@ -44,6 +45,15 @@ public interface ResourceReference {
      */
     InputStream read() throws IOException;
 
+    /**
+     * @return an InputStreamReader to read the contents of the resource
+     *
+     * @throws IOException if unable to obtain an InputStream from the resource
+     */
+    default InputStreamReader createReader() throws IOException {
+        return new InputStreamReader(this.read());
+    }
```

Comment:
We should avoid introducing this default method because it defaults to the platform character encoding.